File: PROGRAMMERS/PYTHON/Lv.1/덧칠하기.py
def solution(n, m, section):
    answer = 0
    paint = [0 for _ in range(n)]

    for i in section:
        paint[i - 1] = 1  # index 1,2,5 칠해짐

    for y in range(0, n - m + 1, m):  # m 단위로
        if paint[y:y + m].count(1) > 0:  # 1이 있다면
            paint[y:y + m] = [0 for _ in range(m)]
            answer += 1

    return answer

File: 프로그래머스/1/161989. 덧칠하기/덧칠하기.java
class Solution {
    public int solution(int n, int m, int[] section) {
        int answer = 0;
        
        int[] wallpaper = new int[n];
        for(int s : section) {
            wallpaper[s-1] = 1;
        }
        
        for(int i = 0; i < n; i++){
            if(wallpaper[i] == 1) {
                // paint
                for(int j = i; j < i+m; j++){
                    if(j >= n){
                        break;
                    }
                    wallpaper[j] = 0;
                }
                answer+=1;
            }
        }
        
        
        return answer;
    }
}